go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -90,6 +90,11 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 	for _, envVar := range envVars {
 		sparkEnvVars[envVar.Name] = envVar.Value
 	}
+
+	for k, v := range config.GetK8sPluginConfig().DefaultEnvVarsFromEnv {
+		sparkEnvVars[k] = v
+	}
```

Comment:
I think the `DefaultEnvVarsFromEnv` are already carried over in the `DecorateEnvVars` [function above](https://github.com/flyteorg/flyteplugins/blob/eed7326340039dbdfcfae75ae6911219ff49371d/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go#L125-L128).

Reply:
Good catch, thanks! I had misunderstood `DefaultEnvVarsFromEnv`. I removed the logic I had added to carry them over again and corrected the test I had added in [this](https://github.com/flyteorg/flyteplugins/pull/271/commits/63fd1820e310c9dca3aff9b3cc257293fb291773) commit.